frontend/src/MainView/index.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { View } from 'react-native';
+import { useSelector } from 'react-redux';
+import Troops from '../Troops';
+import Leaderboard from '../Leaderboard';
+import Buildings from '../Buildings';
+
+const MainViewstyle = {
+  paddingTop: 10,
+};
+
+function MainView() {
+  const currentlyDisplayComponent = useSelector((state) => state.menu.currentlyDisplayComponent);
+  let componentToRender;
+  switch (currentlyDisplayComponent) {
+    case 'Buildings':
+      componentToRender = <Buildings />;
+      break;
+    case 'Troops':
+      componentToRender = <Troops />;
+      break;
+    case 'Battle':
+      componentToRender = <Troops />;
+      break;
+    case 'Leaderboard':
+      componentToRender = <Leaderboard />;
+      break;
+    default:
+      throw new Error('invalid currentlyDisplayComponent');
```

Comment:
do you want to show this to the user? please return with a component which says 'Sorry, something went wrong' or something, but don't throw an error here